//-----------------------------EMBEDDING COMMENTS--------------------------


//POST
postsRouter.post("/:id", async (req, res, next) => {
    try {
    } catch (error) {
      next(error)
    }
  })

//GET
postsRouter.get("/:id/comments", async (req, res, next) => {
    try {
    } catch (error) {
      next(error)
    }
  })

//GET byID
postsRouter.get("/:id/comments/:commentId", async (req, res, next) => {
    try {
    } catch (error) {
      next(error)
    }
  })

//PUT
postsRouter.get("/:id/comment/:commentId", async (req, res, next) => {
    try {
    } catch (error) {
      next(error)
    }
  })

//DELETE
postsRouter.get("/:id/comment/:commentId", async (req, res, next) => {
    try {
    } catch (error) {
      next(error)
    }
  })

  ----------------------------------------------------------------------------------------------------
  what error is this?
  TypeError: post.comments.findById is not a function
    at file:///C:/Users/erliz/OneDrive/Desktop/Homework/ReactJS/strive-homework-m5-w2-p2-bookstore/strive-homework-m6-w2-p2/src/services/posts/index.js:154:42
    at processTicksAndRejections (node:internal/process/task_queues:96:5)

    that was because i typed 
    const theComment = post.comments.find(req.params.commentId)

    instead of:
    const theComment = post.comments.find(comment => comment._id.toString() === req.params.commentId) 

-------------------------------------------------------------------------------------------------------

Error at postman that not stop sending request
although the file is uploaded already to cloudinary
this happened because I forgot to send response back after the uploading process is done

before:
postsRouter.post("/:id/upload", cloudinaryUploader, async(req,res,next)=>{
    try {
        let targetBlogPost = await postsModel.findById(req.params.id)
        if(!targetBlogPost) next(createError(404, `Post with id ${req.params.id} not found!`))
        await targetBlogPost.update({cover:req.file.path})
       
    } catch (error) {
        next(error)
    }
})

after:
postsRouter.post("/:id/upload", cloudinaryUploader, async(req,res,next)=>{
    try {
        let targetBlogPost = await postsModel.findById(req.params.id)
        if(!targetBlogPost) next(createError(404, `Post with id ${req.params.id} not found!`))
        await targetBlogPost.update({cover:req.file.path})

        res.send()  //this one

    } catch (error) {
        next(error)
    }
})